### PR TITLE
feat: update submission state

### PR DESCRIPTION
**Pull request by niall-shaw**
As an administrator, approve / reject a pending submission, and automatically notify the user.

Add an approved submission to the entities list.

### CODE REVIEW

File: packages/frontend/src/app/submissions/[id]/components/ApproveButton.tsx
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/[id]/components/ApproveButton.tsx:1]: A component should not have a dependency on API calls.

File: packages/frontend/src/app/submissions/page.tsx [hunk 1]
```diff
@@ -51,48 +54,52 @@ export default function SubmissionsPage() {
                 !selectedFilters.length || selectedFilters.includes(item.state)
               )
             })
-            .map((item: SubmissionWithEmail, rowIndex: number) => {
+            .map((submission: SubmissionWithEmail) => {
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/page.tsx:57]: Nice refactor 👍

File: packages/frontend/src/common/components/navigation/Header.tsx [hunk 2]
```diff
@@ -2,8 +2,28 @@ import React from 'react'
 import Link from 'next/link'
 import Image from 'next/image'
 import { NavigationItem } from '@/common/components/navigation/NavigationItem'
+import { betterFetch } from '@/api'
+import { getAuthToken } from '@/common/helpers'
 
-export function Header() {
+async function getPendingSubmissionsCount(): Promise<number> {
```
Comment (jakubkoci) [inline packages/frontend/src/common/components/navigation/Header.tsx:8]: Again, I'm really not sure about calling backend directly from a component 🤔

File: packages/backend/src/server.test.ts [hunk 3]
```diff
@@ -263,6 +263,70 @@ describe('api', () => {
...
+    test('update submission state - approved', async () => {
+      const submissionResult = await post(
+        `http://localhost:${port}/api/submissions`,
+        { ...absaSubmission, invitationId: invitation.id },
+        cookie,
+      )
+      const status = submissionResult.status
+      const submissionResponse = await submissionResult.json()
+      expect(status).toEqual(201)
+      expect(submissionResponse.id).toEqual(expect.any(String))
```
Comment (jakubkoci) [inline packages/backend/src/server.test.ts:275]: Isn't this already tested with some other test related to submission creation?

File: packages/backend/src/server.test.ts [hunk 4]
```diff
@@ -263,6 +263,70 @@ describe('api', () => {
...
+      const rejectionResult = await put(
+        `http://localhost:3000/api/submissions/${response.id}`,
+        { state: 'rejected' },
+        cookie,
+      )
+      const updatedSubmission = await rejectionResult.json()
+      expect(updatedSubmission.state).toEqual('rejected')
+    })
+
+    test('update submission state - unknown', async () => {
```
Comment (jakubkoci) [inline packages/backend/src/server.test.ts:309]: Does it mean that the state is gonna be changed to `uknown`? :) The name suggest that according to other tests names and their behaviour, if I understand them correctly.

**Review by jakubkoci — COMMENTED** [review]
Hi Niall. It seems quite good alrady. I know the PR is still in Draft, but my availability during this day might be limited because of my talk at uni and I already spotted some things that might be good to share with you right away :)

**Comment by niall-shaw** [comment]
Note: we should discuss the approach to protecting this page, since at the moment the `submission/[id]` page is a client component.

File: packages/backend/.eslintrc.js [hunk 5]
```diff
@@ -9,6 +9,10 @@ module.exports = {
   rules: {
     // Type is enforced by callers. Not entirely, but it's good enough.
     '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { ignoreRestSiblings: true },
+    ],
```
Comment (jakubkoci) [inline packages/backend/.eslintrc.js:15]: Isn't it good to have such a rule? Do I understand correctly that this won't complain that `x` is not used, right?
```ts
const { x, y, ...rest } = object
console.log(y, rest)
```

But isn't it good in many cases?

File: packages/frontend/src/app/submissions/[id]/components/LazySubmissionDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
+import Image from 'next/image'
+import { Text2xlBold, TextSmBold } from '@/common/components/Typography'
+import { Submission } from '@/common/interfaces'
+import { Label } from '@/common/components/Label'
+
+export function LazySubmissionDetail({
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/[id]/components/LazySubmissionDetail.tsx:6]: I'm not sure about the name `Lazy...`. Do we have a non-lazy variant? Under the term lazy I would imagine a component that's rendered and then lazyly load its content, but I assume that's not the case here.

File: packages/frontend/src/app/submissions/[id]/components/LazySubmissionDetail.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+          ) : (
+            <div className="h-4 w-28 bg-slate-200 animate-pulse rounded-sm" />
+          )}
+        </div>
+        <div className="flex flex-col gap-1">
+          <Label>Submitter&apos;s Email Address</Label>
+          {submission ? (
+            <TextSmBold>{submission.emailAddress}</TextSmBold>
+          ) : (
+            <div className="h-4 w-28 bg-slate-200 animate-pulse rounded-sm" />
```
Comment (jakubkoci) [inline packages/frontend/src/app/submissions/[id]/components/LazySubmissionDetail.tsx:98]: You have a lot of duplications here.

**Review by jakubkoci — APPROVED** [review]

**Pull request merged**